File: output/ChatGPT/avatar/Java/C++/atcoder_ABC162_D.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int n;
    cin >> n;
    string s;
    cin >> s;
    long long rc = 0;
    long long gc = 0;
    long long bc = 0;
    
    for (int i = 0; i < s.length(); i++) {
        if (s[i] == 'R') {
            rc++;
        } else if (s[i] == 'G') {
            gc++;
        } else {
            bc++;
        }
    }
    
    long long result = rc * gc * bc;
    
    for (int i = 1; i <= n; i++) {
        for (int k = i + 1; k <= n; k++) {
            if ((i + k) % 2 == 0) {
                char is = s[i - 1];
                char ks = s[k - 1];
                char js = s[((i + k) / 2) - 1];
                
                if (is != ks && ks != js && js != is) {
                    result--;
                }
            }
        }
    }
    
    cout << result << endl;
    
    return 0;
}

File: output/ChatGPT/avatar/Java/Python/atcoder_ABC162_D.py
def main():
    n = int(input())
    s = input()
    rc = 0
    gc = 0
    bc = 0
    for i in range(len(s)):
        if s[i] == 'R':
            rc += 1
        elif s[i] == 'G':
            gc += 1
        else:
            bc += 1
    result = rc * gc * bc
    for i in range(1, n):
        for k in range(i+1, n+1):
            if (i + k) % 2 == 0:
                is_ = s[i-1]
                ks = s[k-1]
                js = s[((i + k) // 2)-1]
                if is_ != ks and ks != js and js != is_:
                    result -= 1
    print(result)

if __name__ == "__main__":
    main()